extensions/data-workspace/src/common/interfaces.ts

```diff
@@ -97,4 +97,18 @@ export interface IWorkspaceService {
 	 * Shows confirmation message that the extension host will be restarted and current workspace/file will be closed. If confirmed, the specified workspace will be entered.
 	 */
 	enterWorkspace(workspaceFile: vscode.Uri): Promise<void>;
+
+	/**
+	 * Gets the project actions to be placed on the dashboard toolbar
+	 * @param provider The project provider
+	 */
+	getProjectToolbarActions(provider: IProjectProvider): IProjectAction[];
```

Comment:
you are adding this to workspaceService, but it seems to me this should be a property/method in the project provider,

Reply:
Agreed. Removed it from workspaceService, and present only in project provider now.